### ISSUE TITLE
Removing a group in a field-list in a repeat

### ISSUE THREAD

**Issue by dbemke** [getodk/collect#6155]
#### ODK Collect version
the store version 2024.1.3, the master version ffc5aa9faa5dd4157a8322be4e3310314c091def
#### Android version
10, 14
#### Device used
Redmi 9T, Pixel 7a
#### Problem description
I created a form with a repeat group which contains a text question and a regular group on a field-list. When I go to the hierarchy view while on the field-list part and click the bin icon I remove the whole repeat group. On the screen where I remove I don’t see the question which isn’t in the field-list but I remove the whole repeat – the question and the field-list. I’m not sure if it’s expected?
![repeatQuestionFieldlist](https://github.com/getodk/collect/assets/56479916/175ca140-abb4-4832-9743-f42099dbd0a5)

https://github.com/getodk/collect/assets/56479916/52ba1eef-afce-4e0d-b5d6-1ff174ea2301

#### Steps to reproduce the problem
1. Open the form.
[repeatWithQuestionAndContainingFieldList.xlsx.txt](https://github.com/getodk/collect/files/15434305/repeatWithQuestionAndContainingFieldList.xlsx.txt)
2. Fill the first question.
3. Swipe to the next question (field-list) and fill it in.
4. Tap the arrow to go to the hierarchy view.
5. Tap the bin icon and remove the group.

**Comment by lognaturel** [comment]
This is the expected behavior as far as I know. The delete button is for deleting repeat instances. The confirmation dialog does seem pretty clear to me though I see how it’s confusing because groups and repeats are similar. 

We don’t allow modifying the form structure, though, and groups are a fundamental part of that structure whereas repeat instances are used-created content. Maybe adding a title to the confirmation dialog could help?

**Comment by alyblenkin** [comment]
I agree that changing the title might help because I'm assuming most people aren't reading the body copy and that also adds to the confusion. The first option could get really long but at least it's very clear what's happening. If we did go with the first option the body copy could say something like: 

> "**Remove "title" group and all sub groups**"
> "If you delete this group and all sub groups. Removing this group can not be undone."

or make it clear everything will be deleted 

> "**Remove group and all sub groups**"
> "If you delete "title" group and all sub groups. Removing this group can not be undone. You would have to add the items again."

### Another consideration

I also think the end state is a bit confusing because you're left with a blank screen, which is the right behaviour because you've removed everything, but the next step isn't clear. What if...

- The user returned to the summary/hierarchy view, and we displayed a snackbar that says something like "Items were deleted. Undo" with an option to dismiss? It would be great if they could undo the action, just in case they made a mistake. 

- They stay on the blank screen, but we show the snackbar, so it's clear what has happened.
